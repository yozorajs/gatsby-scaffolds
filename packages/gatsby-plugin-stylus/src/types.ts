/**
 * Plugin options.
 */
export interface GatsbyPluginStylusOptions {
  /**
   * Webpack.Rule for.
   * @default {}
   */
  stylusRule?: any
  /**
   * Webpack.Rule.
   * @default {}
   */
  moduleStylusRule?: any
  /**
   * Whether to generate *.sourcemap.
   * @default false
   */
  shouldUseSourceMap?: boolean
  /**
   * Whether to generate *.d.ts for *.styl files.
   * @default false
   */
  shouldGenerateDts?: boolean
  /**
   * Options for 'css-loader'
   */
  cssLoaderOptions?: any
  /**
   * Options for 'stylus-loader'.
   */
  stylusLoaderOptions?: any
  /**
   * Options for 'postcss-loader'.
   */
  postcssLoaderOptions?: any
}
