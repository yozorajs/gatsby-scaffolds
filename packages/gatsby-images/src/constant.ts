export const EMPTY_ALT = 'YOZORA_EMPTY_ALT'
