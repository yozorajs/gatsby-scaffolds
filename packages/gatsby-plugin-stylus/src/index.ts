export * from './types'
export * from './on-create-webpack-config'
