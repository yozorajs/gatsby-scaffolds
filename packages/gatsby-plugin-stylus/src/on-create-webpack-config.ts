import type { CreateWebpackConfigArgs } from 'gatsby'
import type { GatsbyPluginStylusOptions } from './types'

/**
 * Mutate webpack config to support stylus.
 *
 * @param {*} param0
 * @param {GatsbyPluginStylusOptions} param1
 */
export function onCreateWebpackConfig(
  { stage, loaders, actions }: CreateWebpackConfigArgs,
  {
    stylusRule = {},
    moduleStylusRule = {},
    shouldUseSourceMap = false,
    shouldGenerateDts = false,
    cssLoaderOptions = {},
    stylusLoaderOptions = {},
    postcssLoaderOptions = {},
  }: GatsbyPluginStylusOptions = {},
): void {
  const isSSR = [`develop-html`, `build-html`].includes(stage)

  const stylusLoaders = (shouldModules: boolean): any[] => {
    const modulesOptions = {
      namedExport: false,
      exportLocalsConvention: 'camelCaseOnly',
      ...cssLoaderOptions.modules,
    }

    // Generate *.d.ts for stylus files
    const dtsLoader = shouldModules &&
      shouldGenerateDts && {
        loader: '@teamsupercell/typings-for-css-modules-loader',
      }

    // Process css contents
    const cssLoader = loaders.css({
      esModule: true,
      import: true,
      importLoaders: 1,
      sourceMap: shouldUseSourceMap,
      ...cssLoaderOptions,
      modules: shouldModules ? modulesOptions : false,
    })

    // Processing stylus contents.
    const stylusLoader = {
      loader: 'stylus-loader',
      options: {
        sourceMap: shouldUseSourceMap,
        ...stylusLoaderOptions,
      },
    }

    return [
      !isSSR &&
        loaders.miniCssExtract({
          modules: shouldModules ? { namedExport: modulesOptions.namedExport } : {},
        }),
      dtsLoader,
      cssLoader,
      loaders.postcss(postcssLoaderOptions),
      stylusLoader,
    ].filter(Boolean)
  }

  const stylusRegex = stylusRule.test ?? /\.styl$/
  const moduleStylusRegex = moduleStylusRule.test ?? /\.module\.styl$/
  actions.setWebpackConfig({
    module: {
      rules: [
        {
          exclude: moduleStylusRegex,
          ...stylusRule,
          test: stylusRegex,
          use: stylusLoaders(false),
        },
        {
          ...moduleStylusRule,
          test: moduleStylusRegex,
          use: stylusLoaders(true),
        },
      ],
    },
  })
}
