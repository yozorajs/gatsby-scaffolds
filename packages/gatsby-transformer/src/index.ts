export * from './create-schema-customization'
export * from './on-create-node'
export * from './set-field'
export * from './types'
