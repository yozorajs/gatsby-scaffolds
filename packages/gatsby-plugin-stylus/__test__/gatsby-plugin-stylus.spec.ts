import { onCreateWebpackConfig } from '../src'

describe(`gatsby-plugin-stylus`, () => {
  const actions = {
    setWebpackConfig: jest.fn(),
  }

  // loaders "mocks"
  const loaders = {
    miniCssExtract: (): string => `miniCssExtract`,
    css: (args: any): string => `css(${JSON.stringify(args)})`,
    postcss: (args: any): string => `postcss(${JSON.stringify(args)})`,
    null: (): string => `null`,
  }

  beforeEach(() => {
    actions.setWebpackConfig.mockReset()
  })

  const stylusPlugin = jest.fn().mockReturnValue(`foo`)

  const tests = {
    stages: [`develop`, `build-javascript`, `develop-html`, `build-html`],
    options: {
      'Empty options': {},
      'No options': undefined,
      'Stylus options #1': {
        stylusRule: {
          use: [stylusPlugin()],
        },
        moduleStylusRule: {
          test: /\.custom\.styl$/,
          use: [stylusPlugin()],
          import: [`file.js`, `file2.js`],
        },
      },
      'Stylus options #2': {
        shouldGenerateDts: true,
        shouldUseSourceMap: true,
      },
      'PostCss plugins': {
        postCssPlugins: [`test1`],
      },
      'css-loader use commonjs': {
        cssLoaderOptions: {
          esModule: false,
          modules: {
            namedExport: false,
          },
        },
      },
    },
  }

  tests.stages.forEach(stage => {
    for (const label in tests.options) {
      const options = tests.options[label]
      it(`Stage: ${stage} / ${label}`, () => {
        onCreateWebpackConfig(
          {
            actions,
            loaders,
            stage,
          } as any,
          options,
        )
        expect(actions.setWebpackConfig).toMatchSnapshot()
      })
    }
  })
})
