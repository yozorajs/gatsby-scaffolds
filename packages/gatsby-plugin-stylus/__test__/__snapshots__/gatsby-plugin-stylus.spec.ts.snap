// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-stylus Stage: build-html / Empty options 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-html / No options 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-html / PostCss plugins 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss(["test1"])",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss(["test1"])",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-html / Stylus options #1 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.custom\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "import": [
                "file.js",
                "file2.js",
              ],
              "test": /\\\\\\.custom\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-html / Stylus options #2 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":true,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": true,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                {
                  "loader": "@teamsupercell/typings-for-css-modules-loader",
                },
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":true,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": true,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-html / css-loader use commonjs 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":false,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":false,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-javascript / Empty options 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-javascript / No options 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-javascript / PostCss plugins 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss(["test1"])",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss(["test1"])",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-javascript / Stylus options #1 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.custom\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "import": [
                "file.js",
                "file2.js",
              ],
              "test": /\\\\\\.custom\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-javascript / Stylus options #2 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":true,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": true,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                {
                  "loader": "@teamsupercell/typings-for-css-modules-loader",
                },
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":true,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": true,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-javascript / css-loader use commonjs 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":false,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":false,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop / Empty options 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop / No options 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop / PostCss plugins 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss(["test1"])",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss(["test1"])",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop / Stylus options #1 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.custom\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "import": [
                "file.js",
                "file2.js",
              ],
              "test": /\\\\\\.custom\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop / Stylus options #2 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":true,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": true,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                {
                  "loader": "@teamsupercell/typings-for-css-modules-loader",
                },
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":true,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": true,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop / css-loader use commonjs 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":false,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "miniCssExtract",
                "css({"esModule":false,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop-html / Empty options 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop-html / No options 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop-html / PostCss plugins 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss(["test1"])",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss(["test1"])",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop-html / Stylus options #1 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.custom\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "import": [
                "file.js",
                "file2.js",
              ],
              "test": /\\\\\\.custom\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop-html / Stylus options #2 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":true,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": true,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                {
                  "loader": "@teamsupercell/typings-for-css-modules-loader",
                },
                "css({"esModule":true,"import":true,"importLoaders":1,"sourceMap":true,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": true,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop-html / css-loader use commonjs 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "module": {
          "rules": [
            {
              "exclude": /\\\\\\.module\\\\\\.styl\\$/,
              "test": /\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":false,"import":true,"importLoaders":1,"sourceMap":false,"modules":false})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
            {
              "test": /\\\\\\.module\\\\\\.styl\\$/,
              "use": [
                "css({"esModule":false,"import":true,"importLoaders":1,"sourceMap":false,"modules":{"namedExport":false,"exportLocalsConvention":"camelCaseOnly"}})",
                "postcss({})",
                {
                  "loader": "stylus-loader",
                  "options": {
                    "sourceMap": false,
                  },
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
